classes/core/PKPEventServiceProvider.inc.php

```diff
@@ -28,7 +41,26 @@ class PKPEventServiceProvider extends EventServiceProvider
      * @var array $listen $event => $listeners[]
      * @brief Registering events & listeners, see Illuminate\Events\EventServiceProvider
      */
-    protected $listen = [];
+    protected $listen = [
```

Comment:
I recommend using Event Discovery. One of the recent PRs adds file caching to resolve possible performance issues. The only thing is when testing newly added listener to not forget to clear the cache. Sometimes I use Nate's plugin to clear the cache on every request: https://github.com/NateWr/cacheBuster

Reply:
(listeners use the same type-hinting for event discovery as per Laravel's documentation)

Reply:
The only downside with the Event Discovery it's the extensive Reflection usage. An operation like that multiple times it's slowly.

Also, hard coding the most common events and listeners help other developers to understand the flows.

Reply:
This is documented in Laravel docs and those are grouped in designated places in the code so, I think, should ok from a perspective for others to understand the concept. 

The results of calls to Reflection API, in the case of Event Discovery, are cached and it wouldn't be called on every request.

Reply:
I came across a doubt about this approach of caching the discovered events without [extra commands](https://laravel.com/docs/7.x/events#event-discovery) like `event:list`, `event:clear` and `event:cache`.

We should add those ideas on our CLI tools.

Also, add a flag to enable or not the caching development.